### ISSUE TITLE
2.5.1 Regression, custom encoding fails when combined with Jackson

### ISSUE THREAD

**Issue by douglas-m** [immutables/immutables#714]
This bug was introduced with version 2.5.1 (it works correctly in 2.5.0) and is still present in current latest (2.5.5). The example code below produces a compile error, `Could not generate constructor. Attribute 'innerFoo' does not have default value.`

If I remove `@JsonSerialize`, it works.

```
@Encoding
public final class ListEncoding<T> {
    @Encoding.Impl
    private List<T> impl;

    ListEncoding() {
    }

    @Encoding.Builder
    static final class Builder<T> {
        private List<T> list = new ArrayList<>();

        Builder() {
        }

        @Encoding.Naming(value = "set*")
        @Encoding.Init
        void setValue(T obj) {
            list.add(obj);
        }

        @Encoding.Init
        @Encoding.Copy
        void set(List<T> list) {
            this.list = list;
        }

        @Encoding.Build
        List<T> build() {
            return this.list;
        }
    }
}

@JsonSerialize
@Value.Immutable
@ListEncodingEnabled
public interface Foo {
    List<String> getInnerFoo();
}
```

**Comment by elucash** [comment]
thank you for reporting this, the fix will be released in 2.6.0

**Comment by nezda** [comment]
I was finally getting around to leveraging [#428] and spent part of the afternoon coming to this conclusion - awesome I can use 2.5.0 or the upcoming 2.6.0 to work around.  Thanks - `OptionalIntWithSentinel` is working great for me now!
